src/asciidoctor-extensions-api.js

```diff
@@ -87,12 +95,51 @@ Extensions.unregisterAll = function () {
   this.$unregister_all();
 };
 
+/**
+ * Unregister the specified statically-registered extension groups.
+ *
+ * NOTE Opal cannot delete an entry from a Hash that is indexed by symbol, so
+ * we have to resort to using low-level operations in this method.
+ *
+ * @memberof Extensions
+ */
+Extensions.unregister = function () {
+  var names = Array.isArray(arguments[0]) ? arguments[0] : arguments;
+  var groups = this.groups;
+  var groupNameIdx = {};
+  for (var i = 0, groupSymbolNames = groups.$$keys, l1 = groupSymbolNames.length; i < l1; i++) {
+    var groupSymbolName = groupSymbolNames[i];
+    groupNameIdx[groupSymbolName.toString()] = groupSymbolName;
+  }
+  for (var j = 0, l2 = names.length; j < l2; j++) {
+    var groupStringName = names[j];
+    if (groupStringName in groupNameIdx) Opal.hash_delete(groups, groupNameIdx[groupStringName]);
+  }
+};
+
 /**
  * @namespace
  * @module Extensions/Registry
  */
 var Registry = Extensions.Registry;
 
+/**
+ * @memberof Extensions/Registry
+ */
+Registry.$$proto.getGroups = Extensions.getGroups;
+
+/**
+ * @memberof Extensions/Registry
+ */
+Registry.$$proto.unregisterAll = function () {
+  this.groups = Opal.hash2();
```

Comment:
Can we use `Opal.hash` ? I've searched for `Opal.hash2` and found an old GitHub issue: https://github.com/opal/opal/issues/758 :wink:

Reply:
Technically, Opal.hash and Opal.hash2 are both still supported. The difference is that Opal.hash2 doesn't do any processing on the keys. But in this case, it's empty, so we could technically just use Opal.hash_init().

Reply:
Nope, Opal.hash() is the right thing to use.